front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/enableInteractivity.js

```diff
@@ -310,7 +313,32 @@ const enableInteractivity = (
         const positionLocal = rotate
           ? chart.y.invert(pointer(event, event.currentTarget)[1])
           : chart.x.invert(pointer(event, event.currentTarget)[0]);
+        // const timePositionLocal = interpolateOnPosition;
         const timePositionLocal = interpolateOnPosition(dataSimulation, keyValues, positionLocal);
+        const immediatePositionsValuesForPointer = interpolateOnTime(
+          dataSimulation,
+          keyValues,
+          LIST_VALUES_NAME_SPEED_SPACE,
+          datetime2sec(timePositionLocal)
+        );
+
+        // GEV prepareData func multiply speeds by 3.6. We need to normalize that to make a convenitn pointer update
+        LIST_VALUES_NAME_SPEED_SPACE.forEach((name) => {
+          if (
+            immediatePositionsValuesForPointer[name] &&
+            !Number.isNaN(immediatePositionsValuesForPointer[name]?.time) &&
+            !Number.isNaN(immediatePositionsValuesForPointer[name]?.speed)
```

Comment:
you can use the optional chaining with brackets : `?.[]`

```suggestion
            !Number.isNaN(immediatePositionsValuesForPointer?.[name]?.time) &&
            !Number.isNaN(immediatePositionsValuesForPointer?.[name]?.speed)
```

Reply:
Done